### PR TITLE
For #15348: Voice search option appears for all search engines

### LINKED ISSUE #15348: [Bug] Don't restrict Voice Search to Google search engine, it works with others too
Hi Fenix team, 

I did some testing with Voice Search because I was intrigued to know what works and what doesn't. Fenix seems to use whatever service is set as the "Voice Input" in Settings => Apps and Notifications => Default Apps => Assist and Voice Input => Voice Input.  This need not be Google, but it is very difficult to find other voice input services available. I tried Konele and it worked.

However, Fenix only makes the Voice Search microphone button available when Google is chosen as the default search provider, which makes it appear that Google Voice Search is being used. I am pretty sure this is not the case because I can select another search provider after using the voice search function.

## Steps to reproduce
1. Choose a Voice Input service other than Google in Settings => Apps and Notifications => Default Apps => Assist and Voice Input => Voice Input; I used Konele for this testing
2. Choose Google as the default search provider in Fenix
3. Go to the search page
4. Tap the input button next to the text field
5. Use Konele (or other) to convert speech-to-text and enter text into the search text field
6. Tap Search Engine and select another search provider
7.  Search and get results.

### Expected behaviour
This shouldn't work, because Fenix only allows Voice Search icon if Google is selected as the default search engine, suggesting that it only works with Google services.

### Actual behaviour
It works! Voice Search is not dependent on Google services at all, it just needs something which does speech-to-text installed as the Voice Input service.

### Device information

* Android device: Nokia 8 with Android 9.
* Fenix version: Nightly.

Cheers 🙂

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-7317)

### PR DESCRIPTION
**Pull request by DreVla**
Voice search was restricted only for google, with the condition removed it should appear now for all search engines

https://user-images.githubusercontent.com/32488956/205888210-e94bee42-6136-41a5-adc9-ef0f9fddae8d.mp4

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #15348

### CODE REVIEW

**Comment by DreVla** [comment]
Draft while waiting for approval from product.

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt [hunk 1]
```diff
@@ -765,12 +765,9 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
...
 
-    private fun updateVoiceSearchButton(searchFragmentState: SearchFragmentState) {
-        val searchEngine = searchFragmentState.searchEngineSource.searchEngine
-
+    private fun updateVoiceSearchButton() {
         val isVisible =
-            searchEngine?.id?.contains("google") == true &&
-                isSpeechAvailable() &&
+            isSpeechAvailable() &&
                 requireContext().settings().shouldShowVoiceSearch
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:771]: Let's just inline this and remove the `isVisible` variable.

**Review by gabrielluong — COMMENTED** [review]
I was gonna approve this based on the comments on https://mozilla-hub.atlassian.net/browse/FNXV2-22043. 

So far, it seems like it is a YES from UX, but I held back on the approval since we might want for Product to give a firm YES.

**Review by gabrielluong — APPROVED** [review]
Got the OK from @kbrosnan https://mozilla-hub.atlassian.net/browse/FNXV2-22043?focusedCommentId=624238

**Pull request merged**